### ISSUE TITLE
Build fails if Git repo has no commits

### ISSUE THREAD

**Issue by knowler** [roots/bud#61]
* [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing issue
* [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

```
/Users/knowler/code/testing-bud/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: Command failed with exit code 128: git rev-parse --short HEAD --no-color
fatal: Needed a single revision
    at makeError (/Users/knowler/code/testing-bud/node_modules/execa/lib/error.js:59:11)
    at handlePromise (/Users/knowler/code/testing-bud/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 128: git rev-parse --short HEAD --no-color',
  command: 'git rev-parse --short HEAD --no-color',
  exitCode: 128,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: 'fatal: Needed a single revision',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
error Command failed with exit code 7.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Steps to reproduce

Initialize a new git repo, but don't commit anything. `git init`.

## Versions

You can get this information from referencing `CHANGELOG.md`. Also, please include the OS and what version of the OS you're running.

bud 4.1.0
Node 16
macOS 11.3.1

## Additional information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by kellymears** [comment]
Solution is going to be checking for git presence in the useGit hook 

the status effect should treat this check as a dependency (or otherwise guard itself):

https://github.com/roots/bud/blob/7499b26f647d35ea99b23bcbdb6975a7d2051c94/packages/%40roots/bud-dashboard/src/hooks/useGit.ts#L10-L18

I wonder if there is a git command that won't return an error outside of a git dir or if the best options is to check for/in `.git` directly.

**Comment by knowler** [comment]
If you did the latter, then you'll also want to check for a `.git` file since if it's just a worktree then that's what will be there, instead of a directory. Also, the latter method does not account for the Git directory being called something else.

I might be able to dig up a command that works for the former.

**Comment by kellymears** [comment]
I think I fixed this (just using try/catch). I couldn't find any git commands that don't throw if you are outside of a repo. Same for repos with no commits. So I just wrapped it in try/catch and set up state to track it (as outlined above).

I guess you could check for both with something like [find-up](https://www.npmjs.com/package/find-up) but that seems expensive in comparison to just trying (?)

I'll likely release a patch version (v4.1.1) with the fix from #62 tomorrow but I'll leave this issue open in case someone has a more elegant solution.
